source/Firebase/Functions/ApiDefinition.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+
+using UIKit;
+using Foundation;
+using ObjCRuntime;
+using CoreGraphics;
+using Firebase.Core;
+
+namespace Firebase.Functions
+{
+	// @interface FIRFunctions : NSObject
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSObject), Name = "FIRFunctions")]
+	interface Functions
+	{
+		// + (FIRFunctions *)functions;
+        [Static]
+        [Export("functions")]
+        Functions DefaultInstance { get; }
+
+		// + (FIRFunctions *)functionsForApp:(FIRAPP *)app;
+		[Static]
+		[Export("functionsForApp:")]
+		Functions CreateForApp(App app);
+
+		//+ (FIRFunctions *) functionsForRegion:(NSString*) region;
+        [Static]
+        [Export("functionsForRegion:")]
+		Functions CreateForRegion(string region);
+
+		//+ (FIRFunctions *)functionsForApp:(FIRApp *)app region:(NSString*) region
+        [Static]
+        [Export("functionsForApp:region")]
+		Functions CreateForApp(App app, string region);
+
+        //- (FIRHTTPSCallable *)HTTPSCallableWithName:(NSString *)name;
+        [Export("HTTPSCallableWithName:")]
+		HTTPSCallable HTTPSCallableWithName(string name);
```

Comment:
We use CamelCase for names and remove `With*`, `For*`, `From*` when possible, due to the parameter helps to give extra info of the method.

```suggestion
		HttpsCallable HttpsCallable (string name);
```